### PR TITLE
Making sure middleware and ingressrouth paths are the same

### LINKED ISSUE #959: HLD Reconcile: generated middleware strips the wrong path
**Describe the bug:**

A generated middleware yaml from `hld reconcile` does not strip the full path that maps to the Ingress Route

**To Reproduce:**

Using this bedrock.yaml:

```
rings:
  master:
    isDefault: true
services:
  ./services/fabrikam:
    displayName: 'fabrikam'
    k8sBackendPort: 80
    k8sBackend: 'fabrikam-k8s-svc'
    pathPrefix: 'fabrikam-service'
    pathPrefixMajorVersion: 'v1'
    helm:
      chart:
        branch: master
        git: 'https://dev.azure.com/foo/bar/_git'
        path: stable/wordpress
    middlewares:
      - ''
variableGroups:
  - fabrikam-vg
```

Running `spk hld reconcile` on an application repository produces this middleware (note the prefixes):

```
apiVersion: traefik.containo.us/v1alpha1
kind: Middleware
metadata:
  name: fabrikam-master
spec:
  stripPrefix:
    forceSlash: false
    prefixes:
      - /fabrikam
```

and this ingress route (note the path path match):

```
apiVersion: traefik.containo.us/v1alpha1
kind: IngressRoute
metadata:
  name: fabrikam-master
spec:
  routes:
    - kind: Rule
      match: 'PathPrefix(`/v1/fabrikam-service`) && Headers(`Ring`, `master`)'
      middlewares:
        - name: fabrikam-master
      services:
        - name: fabrikam-k8s-svc
          port: 80
```

**Expected behavior:**

The generated middleware to utilize the path prefix and the pathPrefixMajorVersion as a prefix strip ie:

```
apiVersion: traefik.containo.us/v1alpha1
kind: Middleware
metadata:
  name: fabrikam-master
spec:
  stripPrefix:
    forceSlash: false
    prefixes:
      - /v1/fabrikam-service
```

**Additional context:**

It looks like the middleware presently uses just the display name of the service

**Does this require updates to documentation?:**

No

### PR DESCRIPTION
**Pull request by mtarng**
Closes https://github.com/microsoft/bedrock/issues/959

### CODE REVIEW

File: src/commands/hld/reconcile.ts [hunk 1]
```diff
@@ -309,25 +311,51 @@ export const reconcileHld = async (
...
+ */
+export const getFullPathPrefix = (
+  majorVersion: string,
+  pathPrefix: string,
+  serviceName: string
+): string => {
+  const versionPath = majorVersion ? `/${majorVersion}` : "/";
+  const servicePath = pathPrefix || serviceName;
+
+  return path.join(versionPath, servicePath);
```
Comment (dennisseah) [inline src/commands/hld/reconcile.ts]: I thought this should be
```
  const versionPath = majorVersion ? `/${majorVersion}` : "";
  const servicePath = pathPrefix || serviceName;
  return `${versionPath}/${servicePath}`;
```
this is URL path and not file system path (if i am not wrong)

Comment (bnookala) [reply src/commands/hld/reconcile.ts]: @dennisseah I think @mtarng's fix should be fine - we're specifying the full path to strip (from the root) of the url ie:

http://bar.com/v1/fabrikam should strip /v1/fabrikam

Comment (bnookala) [reply src/commands/hld/reconcile.ts]: and even then:

```
> path.join("/", "foo")
'/foo'
```

**Review by bnookala — APPROVED** [review]

File: src/commands/hld/reconcile.ts (hunk 1 above)
Comment (dennisseah) [reply src/commands/hld/reconcile.ts]: `path` library should not be used on constructing URL. separator is always `/` is there a reason for using `path`? Thanks

**Review by yradsmikham — APPROVED** [review]

File: src/commands/hld/reconcile.ts (hunk 1 above)
Comment (dennisseah) [reply src/commands/hld/reconcile.ts]: https://stackoverflow.com/questions/16301503/can-i-use-requirepath-join-to-safely-concatenate-urls

**Review by edaena — APPROVED** [review]

**Pull request merged**